File: src/main/java/com/redgrapefruit/mythicaltowers/common/armor/BootsItem.java
package com.redgrapefruit.mythicaltowers.common.armor;

import com.redgrapefruit.mythicaltowers.common.init.MythicalItemGroups;
import net.minecraft.entity.EquipmentSlot;
import net.minecraft.item.ArmorItem;
import net.minecraft.item.ArmorMaterial;
import net.minecraft.item.Item;

/**
 * Custom boots with special effects when put on
 */
public class BootsItem extends ArmorItem {
    public BootsItem(ArmorMaterial material) {
        super(material, EquipmentSlot.FEET, new Item.Settings().group(MythicalItemGroups.ARMOR));
    }
}

File: src/main/kotlin/com/redgrapefruit/mythicaltowers/common/armor/BootsItem.kt
package com.redgrapefruit.mythicaltowers.common.armor

import com.redgrapefruit.mythicaltowers.common.init.MythicalItemGroups
import net.minecraft.entity.EquipmentSlot
import net.minecraft.entity.effect.StatusEffect
import net.minecraft.item.ArmorItem
import net.minecraft.item.ArmorMaterial

/**
 * Custom boots with special effects when put on
 */
class BootsItem(
    material: ArmorMaterial,
    val effect: StatusEffect,
    val amplifier: Int
) : ArmorItem(material, EquipmentSlot.FEET, Settings().group(MythicalItemGroups.ARMOR))